Sources/MessagingInApp/hooks/HookProvider.swift

```diff
@@ -11,12 +11,14 @@ class MessagingInAppModuleHookProvider: ModuleHookProvider {
     var profileIdentifyHook: ProfileIdentifyHook? {
         guard let diGraph = diGraph else { return nil }
 
-        return MessagingInAppImplementation(diGraph: diGraph)
+        // FIXME: [CDP] Find workaround or reuse existing instance to utilize customer provided siteid
```

Comment:
Idea that may or may not work: Would the journey's module replace the need for hooks for us? 

If the journeys module allows the in-app module to be a consumer, the in-app module can get notified when a profile is identified or when a screen is tracked. Therefore, replacing what hooks is doing for us. 

I wanted to share this if it's helpful to avoid this possibly complex `FIXME` scenario.

Reply:
Honestly, I haven't given this much thought yet. But I agree that Journey being the bridge for feature modules communication with tracking/CDP would likely be the better choice for setting up hooks.

For this PR, I believe the only action item is to update the `FIXME` comment. We can then evaluate and make better decision once the Journey module is available, giving us a clearer picture.